library/xml/src/modules/XML.rb

```diff
@@ -247,7 +247,13 @@ def parse_node(node, result)
       children = children.select(&:element?)
       # we need just direct text under node. Can be splitted with another elements
       # but remove whitespace only text
-      text = node.xpath("text()").text.strip
+      #
+      # Do NOT strip trailing white space in CDATA blocks. Maybe people put
+      # it intentionally there (bsc#1195910).
+      #
+      # See also #add_element.
+      text_nodes = node.xpath("text()")
+      text = text_nodes.map { |n| n.cdata? ? n.content.lstrip : n.content.strip }.join
```

Comment:
well, to be on safe side, I would do not stripping on cdata and below when writting checking spaces also at the beginning. This would make our to_xml/from_xml much more consistent as we never loose some characters that user can intentionally place.

Reply:
As said before, this exaple will not work if we keep leading white space:
https://doc.opensuse.org/projects/autoyast/#id-1.9.4.2.4.5